code/src/gfx.c

```diff
@@ -376,37 +392,37 @@ static void Gfx_DrawSpoilerItemGroups(void) {
 }
 
 static void Gfx_DrawERTracker(void) {
-    if (gSettingsContext.shuffleOverworldEntrances || gSettingsContext.shuffleDungeonEntrances) {
-        u16 itemCount = ENTRANCE_OVERRIDES_MAX_COUNT;
+    if (gEntranceTrackingData.EntrancePairsCount > 0) {
+        u16 itemCount = gEntranceTrackingData.EntrancePairsCount;
         u16 firstItem = entranceScroll + 1;
         u16 lastItem = entranceScroll + MAX_ITEM_LINES;
         if (lastItem > itemCount) { lastItem = itemCount; }
         Draw_DrawFormattedString(10, 10, COLOR_TITLE, "Randomized Entrances (%d - %d) / %d",
             firstItem, lastItem, itemCount);
 
-        // TODO random entrances visited %
-        // Draw_DrawFormattedString(SCREEN_BOT_WIDTH - 10 - (SPACING_X * 6), 10, itemPercent == 100 ? COLOR_GREEN : COLOR_WHITE, "%5.1f%%", itemPercent);
+        Draw_DrawFormattedString(SCREEN_BOT_WIDTH - 10 - (SPACING_X * 6), 10, entrancesPercent == 100 ? COLOR_GREEN : COLOR_WHITE, "%5.1f%%", entrancesPercent);
 
         u16 listTopY = 26;
         for (u32 item = 0; item < MAX_ITEM_LINES; ++item) {
             u32 locIndex = item + entranceScroll;
             if (locIndex >= itemCount) { break; }
 
-            EntranceOverride entrOverride = rEntranceOverrides[locIndex];
-            if (entrOverride.index == 0 && entrOverride.override == 0) {
-                continue;
-            }
-
             u32 locPosY = listTopY + ((SPACING_SMALL_Y + 1) * item * 2);
             u32 itemPosY = locPosY + SPACING_SMALL_Y;
-            bool isDiscovered = SaveFile_GetIsEntranceDiscovered(entrOverride.index);
+
+            EntranceTrackingPair pair = gEntranceTrackingData.EntrancePairs[locIndex];
+            bool isDiscovered = IsEntrancePairDiscovered(pair);
+
             u32 color = isDiscovered ? COLOR_GREEN : COLOR_WHITE;
-            Draw_DrawFormattedString_Small(10, locPosY, color,
-                "0x%04X -> 0x%04X, BW: 0x%04X", entrOverride.index, entrOverride.override, entrOverride.blueWarp);
+            const char* unknown = "???";
+            const char* startName = pair.StartStrOffset != ENTRANCE_INVALID_STRING_OFFSET ? &gEntranceTrackingData.StringData[pair.StartStrOffset] : "START STRING NOT FOUND";
+            const char* returnName = pair.ReturnStrOffset != ENTRANCE_INVALID_STRING_OFFSET ? &gEntranceTrackingData.StringData[pair.ReturnStrOffset] : "RETURN STRING NOT FOUND";
+
+            Draw_DrawFormattedString_Small(10, locPosY, color, "%s %c", isDiscovered ? startName : unknown, H_DOUBLE_ARROW_CHR);
+            Draw_DrawFormattedString_Small(10, itemPosY, color, "  %s", isDiscovered ? returnName : unknown);

```

Comment:
```suggestion
            Draw_DrawFormattedString_Small(10, locPosY, color, "%s %c", gSettingsContext.ingameSpoilers || isDiscovered ? startName : unknown, H_DOUBLE_ARROW_CHR);
            Draw_DrawFormattedString_Small(10, itemPosY, color, "  %s", gSettingsContext.ingameSpoilers || isDiscovered ? returnName : unknown);
```
I forgot about this yesterday.